### PR TITLE
Hl 74: api documentation

### PR DESCRIPTION
**Pull request by haavikko**
## Description :sparkles:

* Add API documentation for application/company 
* Additional dependencies required for api documentation: pyyaml, uritemplate, drf-spectacular, django-filter
* Copied&modified a bit instructions in README.md from other city of helsinki project
* Changed application status filtering to use django-filter, so that schema can be generated automatically

### CODE REVIEW

File: backend/benefit/README.md [hunk 1]
```diff
@@ -54,6 +54,54 @@ The project is now running at [localhost:8000](http://localhost:8000)
     * `pip-sync requirements.txt requirements-dev.txt`
 
 
+## Documentation
+
+API documentation automatically generated from OpenAPI schema can be found from [http://127.0.0.1:8000/api_docs/](http://127.0.0.1:8000/api_docs/).
```
Comment (klemmari1) [inline backend/benefit/README.md:59]: This url gives 404: http://127.0.0.1:8000/api_docs/

**Review by klemmari1 — COMMENTED** [review]
The commits can be squashed together and there are some conflicts with develop branch.

File: backend/benefit/README.md (hunk 1 above)
Comment (haavikko) [reply backend/benefit/README.md]: fixed

File: backend/benefit/README.md [hunk 2]
```diff
@@ -54,6 +54,52 @@ The project is now running at [localhost:8000](http://localhost:8000)
     * `pip-sync requirements.txt requirements-dev.txt`
 
 
+## Documentation
+
+The OpenAPI schema is served from [http://127.0.0.1:8000/openapi/](http://127.0.0.1:8000/openapi/).
+
+Swagger documentation can be found at [http://localhost:8000/api_docs/swagger/](http://localhost:8000/api_docs/swagger/)
+and redoc documentation at [http://localhost:8001/api_docs/redoc/](http://localhost:8001/api_docs/redoc/)
```
Comment (klemmari1) [inline backend/benefit/README.md:62]: 8001 -> 8000

**Review by klemmari1 — APPROVED** [review]
One small fix. Otherwise looks good to me!

File: backend/benefit/README.md (hunk 2 above)
Comment (haavikko) [reply backend/benefit/README.md]: Thanks, fixed that too

**Pull request merged**